packages/backend-core/src/sql/sql.ts

```diff
@@ -113,34 +114,45 @@ function generateSelectStatement(
   knex: Knex
 ): (string | Knex.Raw)[] | "*" {
   const { resource, meta } = json
+  const client = knex.client.config.client as SqlClient
 
   if (!resource || !resource.fields || resource.fields.length === 0) {
     return "*"
   }
 
-  const schema = meta?.table?.schema
+  const schema = meta.table.schema
   return resource.fields.map(field => {
-    const fieldNames = field.split(/\./g)
-    const tableName = fieldNames[0]
-    const columnName = fieldNames[1]
-    const columnSchema = schema?.[columnName]
-    if (columnSchema && knex.client.config.client === SqlClient.POSTGRES) {
-      const externalType = schema[columnName].externalType
-      if (externalType?.includes("money")) {
-        return knex.raw(
-          `"${tableName}"."${columnName}"::money::numeric as "${field}"`
-        )
-      }
+    const [table, column, ...rest] = field.split(/\./g)
+    if (
+      client === SqlClient.POSTGRES &&
+      schema[column].externalType?.includes("money")
+    ) {
+      return knex.raw(`"${table}"."${column}"::money::numeric as "${field}"`)
     }
     if (
-      knex.client.config.client === SqlClient.MS_SQL &&
-      columnSchema?.type === FieldType.DATETIME &&
-      columnSchema.timeOnly
+      client === SqlClient.MS_SQL &&
+      schema[column]?.type === FieldType.DATETIME &&
+      schema[column].timeOnly
     ) {
-      // Time gets returned as timestamp from mssql, not matching the expected HH:mm format
+      // Time gets returned as timestamp from mssql, not matching the expected
+      // HH:mm format
       return knex.raw(`CONVERT(varchar, ${field}, 108) as "${field}"`)
     }
-    return `${field} as ${field}`
+
+    // There's at least two edge cases being handled in the expression below.
+    //  1. The column name could start/end with a space, and in that case we
+    //     want to preseve that space.
+    //  2. Almost all column names are specified in the form table.column, except
+    //     in the case of relationships, where it's table.doc1.column. In that
+    //     case, we want to split it into `table`.`doc1.column` for reasons that
+    //     aren't actually clear to me, but `table`.`doc1` breaks things with the
+    //     sample data tests.
+    return knex.raw(
+      `${quote(client, table)}.${quote(
+        client,
+        [column, ...rest].join(".")
+      )} as ${quote(client, field)}`
+    )
```

Comment:
This is the actual fix. The previous `${field} as ${field}` gets passed to `knex.select(...)` which, when `as` is present, parses the string.

So you get the following:

```typescript
knex.select("foo as bar")  // `foo` as `bar`
knex.select("foo  as bar") // `foo` as `bar`
knex.select(" foo as bar") // `foo` as `bar`
```

This change forces the spaces to preserved.